common/src/com/thoughtworks/go/domain/materials/git/GitCommand.java

```diff
@@ -59,7 +59,11 @@ public int cloneFrom(ProcessOutputStreamConsumer outputStreamConsumer, String ur
     // Clone repository from url with specified depth.
     // Special depth 2147483647 (Integer.MAX_VALUE) are treated as full clone
     public int cloneFrom(ProcessOutputStreamConsumer outputStreamConsumer, String url, Integer depth) {
-        CommandLine gitClone = git().withArg("clone").withArg(String.format("--branch=%s", branch));
+        CommandLine gitClone = git()
```

Comment:
This feels a bit odd in that it claims to say clone, and does not behave like `git clone` can we make the `noCheckout` a boolean arg?

Also, don't we need a clone with checkout on the agent?